modules/python/test/test_misc.py

```diff
@@ -50,39 +68,575 @@ class Arguments(NewOpenCVTests):
+                                 "Expected: {2} vs Actual: {3}".format(
+                                     convertible,
+                                     type(convertible).__name__,
+                                     expected, actual
+                                 ))
+
+        for not_convertible in (np.array([123, ], dtype=np.uint8),):
+            with self.assertRaises(
+                (TypeError, OverflowError),
+                msg='Exception is not risen for {} of type {}'.format(
+                    not_convertible, type(not_convertible).__name__
+                )
+            ):
+                _ = cv.utils.dumpSizeT(not_convertible)
+
+    def test_parse_to_float(self):
+        min_float, max_float = get_limits(ctypes.c_float)
+        for convertible in (2, True, False, -13, 1.24, float(32),
+                            np.float(32.45), np.double(12.23),
+                            np.float32(12.3), min_float, max_float, np.inf,
+                            -np.inf, float('Inf'), -float('Inf'),
+                            np.double(np.inf), np.double(-np.inf),
+                            np.double(float('Inf')), np.double(-float('Inf'))):
+            expected = 'Float: {0:.2f}'.format(convertible).lower()
+            try:
+                actual = cv.utils.dumpFloat(convertible).lower()
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, convertible,
+                        type(convertible).__name__
+                    )
+                )
+            else:
+                self.assertEqual(expected, actual,
+                                 msg="Can't convert {0} of type {1}\n"
+                                 "Expected: {2} vs Actual: {3}".format(
+                                     convertible,
+                                     type(convertible).__name__,
+                                     expected, actual
+                                 ))
+
+        # Workaround for Windows NaN tests due to Visual C runtime
+        # special floating point values (indefinite NaN)
+        for nan in (float('NaN'), np.nan, np.float32(np.nan), np.double(np.nan),
+                    np.double(float('NaN'))):
+            try:
+                res = cv.utils.dumpFloat(nan).lower()
+                self.assertIn('nan', res,
+                              msg="Can't convert nan to float: {}".format(res))
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, nan,
+                        type(nan).__name__
+                    )
+                )
+
+        min_double, max_double = get_limits(ctypes.c_double)
+        for inf in (min_float * 10, max_float * 10, min_double, max_double):
```

Comment:
Why is `* 10` needed?

Reply:
To check that if mantissa exceeds max value that can be held by float type it becomes inf.

Maybe this check is unnecessary, because max/min _double checks the same case.